docs/source/developer/architecture/backend/grocery_service.rst

```diff
@@ -0,0 +1,128 @@
+Grocery Service
+================
+
+A data service specializing in loading workspace data from disk into mantid's ADS, for use inside algorithms.
```

Comment:
Can you add links to the [glossary](https://snapred--266.org.readthedocs.build/en/266/glossary.html#term-Service-Component) like 
```suggestion
A :term:`Data Component`specializing in loading :term:`Workspace` data from disk into :term:`Mantid`'s :term:`ADS`, for use inside :term:`Algorithm`s.
```
Note that Workspace, Mantid, and ADS are not currently in the glossary.